### PR TITLE
Vim emulation: fix 0 command

### PR DESCRIPTION
**Pull request by mcclurgm**
Part of splitting #633. Based on the implementation of the `I` command and gnome-builder.

Note, this does not work with selection. It simply removes the selection entirely. But as far as I can tell, the plugin does not implement selection in any way, so I don't believe this is a regression.

### CODE REVIEW

**Comment by jeremypw** [comment]
This seems to move the cursor down a line - not to the beginning of the line (which I think is what is expected from https://devhints.io/vim).

File: plugins/vim-emulation/vim-emulation.vala [hunk 1]
```diff
@@ -212,7 +212,11 @@ public class Scratch.Plugins.VimEmulation : Peas.ExtensionBase, Peas.Activatable
             case Gdk.Key.Home:
             case Gdk.Key.@0:
                 if (number == "") {
-                    view.move_cursor (Gtk.MovementStep.DISPLAY_LINES, 1, false);
+                    var buffer = view.buffer;
+                    Gtk.TextIter insert;
+                    buffer.get_iter_at_mark (out insert, buffer.get_insert ());
+                    insert.set_line_offset (0);
+                    buffer.place_cursor (insert);
```
Comment (jeremypw) [inline plugins/vim-emulation/vim-emulation.vala:219]: This can simply be `view.move_cursor (Gtk.MovementStep.DISPLAY_LINE_ENDS, -1, false);`

**Review by jeremypw — CHANGES_REQUESTED** [review]
See inline comment

File: plugins/vim-emulation/vim-emulation.vala (hunk 1 above)
Comment (mcclurgm) [reply plugins/vim-emulation/vim-emulation.vala:219]: That seems like it should, but it ignores whitespace, so it moves to after any indentation. `0` is supposed to move the the first character of the line, regardless of indentation. That's one of the issues I was trying to fix with this PR. The whitespace-respecting command is `^`.

Comment (jeremypw) [reply plugins/vim-emulation/vim-emulation.vala:219]: Oh, yeah - sorry.  In that case you can fix it with fewer lines with:
```
                    view.move_cursor (Gtk.MovementStep.PARAGRAPH_ENDS, -1, false);
                    view.move_cursor (Gtk.MovementStep.DISPLAY_LINE_ENDS, -1, false);
```

This seems to give the same behaviour as the Vim app.

Comment (mcclurgm) [reply plugins/vim-emulation/vim-emulation.vala:219]: Hmm, looks good in my testing. Makes sense with the docs too. That's a nice improvement, thanks!

**Review by jeremypw — APPROVED** [review]
Lets go!

**Comment by jeremypw** [comment]
Need someone to fix the flatpak CI before this can be merged.

**Pull request merged**